docs/source/migration.md

```diff
@@ -0,0 +1,138 @@
+
+## Introspection
+
+In the Apollo CLI, introspection of running services happens behind the scenes when you pass an `--endpoint` to the `service:push`, `service:check` and `service:download` commands. In Rover, you need to run the `graph introspect` command and pipe the result of that to the similar `graph/subgraph push` and `graph/subgraph check` commands.
+
+The intention behind adding this step was to make introspection more flexible and transparent when it happens in Rover. In addition, separating introspection into its own step should make it more debuggable if you run into errors.
+
+If your goal is to download a schema from a running endpoint, you can output the result of `graph introspect` straight to a file. You can find more about how stdin/stdout works with Rover [here](./essentials#io).
+
+## Monolithic vs. Federated Graphs
+
+The Apollo CLI's API uses a single command set for working with monolithic and federated graphs, the `apollo service:*` commands. Rover treats graphs and federated graphs separately. Typically, when working with federated graphs in Rover, you will be using the `subgraph` command set, since most operations on federated graphs are interacting with a federated graph's subgraphs (ex. checking changes to a single subgraph). Additionally, Rover has an idea of a `supergraph`, which is used for building and working with supergraph schemas.
+
+For more info on the difference in `graph`s, `subgraph`s and `supergraph`s, see [Conventions](./essentials#graph--subgraph--supergraph).
+
+## Limitations of Rover
+
+Not all of the Apollo CLI's features have a compatible feature in Rover, but many of the most popular features are supported in Rover.
+
+For the sake of maintainability, roadmap considerations, and time, Rover has intentionally left off some familiar features of the Apollo CLI.
+
+### Client commands
+
+Rover's focus is around providing an excellent graph management experience. For that reason, we have intentionally left out commands that may be useful for developers working on client projects like `client:codegen` and `client:check`.
+
+### Globs
+
+The Apollo CLI used globs extensively to support using multiple local files and even automatic discovery of files in a directory tree. While this was helpful in a lot of cases, the globbing strategy in the Apollo CLI resulted in a lot of issues and confusion. Rover has intentionally left globs off of file specifiers for initial release.
+
+As a workaround, you may be able to use `cat` to combine multiple files, and pass them to Rover with [stdin](./essentials#io)
+
+### Machine-readable output
+
+In the Apollo CLI, many commands supported alternate output formatting options like `--json` and `--markdown`. We do intend on adding structured output that can be machine readable, but Rover currently does not support it.
```

Comment:
> We do intend on adding structured output that can be machine readable, but Rover currently does not support it.

^ i'm a bit wary of mentioning future plans in a doc like this as it can lead to disappointment if it takes too long or never really happens. i think we'll definitely do `--json` but not as sure about `--markdown`

```suggestion
In the Apollo CLI, many commands supported alternate output formatting options like `--json` and `--markdown`. This is not currently supported by Rover.
```